### PR TITLE
Logs extension

### PR DESCRIPTION
**Pull request by alonre24**
**Describe the changes in the pull request**

Add log level to our log messages and callback. Use "verbose" for the existing logs and add some more messages regarding capacity changes. + tiny fix in a test to reduce flakiness.

### CODE REVIEW

File: src/VecSim/vec_sim_tiered_index.h [hunk 1]
```diff
@@ -53,6 +53,10 @@ class VecSimTieredIndex : public VecSimIndexInterface {
         this->SubmitJobsToQueue(this->jobQueue, this->jobQueueCtx, jobs.data(), callbacks.data(),
                                 jobs.size());
     }
+    template <typename... Args>
+    void log(const char *level, const char *msg, Args... args) const {
+        this->backendIndex->log(level, msg, args...);
+    }
```
Comment (meiravgri) [inline src/VecSim/vec_sim_tiered_index.h:59]: So for each call to log, we will have an additional function symbol in the binary file 
for example: ("%s, %s", "meow", "charlie") and ("%s",meow) yields 2 symbols
I suggest that the tiered index will hold a copy to the backend index log function

File: tests/unit/test_common.cpp [hunk 2]
```diff
@@ -464,12 +464,12 @@ TEST(CommonAPITest, testlog) {
         dynamic_cast<BruteForceIndex<float, float> *>(BruteForceFactory::NewIndex(&params));
```
Comment (meiravgri) [inline tests/unit/test_common.cpp:464]: consider to add a a test also for tierred index

File: tests/unit/test_hnsw_tiered.cpp [hunk 3]
```diff
@@ -2889,7 +2889,7 @@ TYPED_TEST(HNSWTieredIndexTest, switchWriteModes) {
                          .metric = VecSimMetric_L2,
                          .multi = TypeParam::isMulti(),
                          .M = 32,
-                         .efRuntime = 2 * n};
+                         .efRuntime = 3 * n};
```
Comment (meiravgri) [inline tests/unit/test_hnsw_tiered.cpp:2892]: why the change?

**Review by meiravgri — COMMENTED** [review]
See comment below :)

File: tests/unit/test_hnsw_tiered.cpp (hunk 3 above)
Comment (alonre24) [reply tests/unit/test_hnsw_tiered.cpp:2892]: see https://redis.slack.com/archives/C03QDNA8K0B/p1690090714770389

**Review by meiravgri — APPROVED** [review]

**Review by DvirDukhan — APPROVED** [review]
